force-app/main/default/classes/RecurringDonationsController.cls

```diff
@@ -0,0 +1,80 @@
+/*
+    Copyright (c) 2022, Salesforce.org
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2022
+* @group Donation
+* @description Controller class for Recurring Donations LWC.
+*/
+public with sharing class RecurringDonationsController {
+    
+    @TestVisible
+    private static RecurringDonationsService donationService {
+        get {
+            if (donationService == null) {
+                donationService = new RecurringDonationsService();
+            }
+            return donationService;
+        }
+        set {
+            donationService = value;
+        }
+    }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2022
+    * @description Recurring Donations inner class.
+    */
+    public class RecurringDonation {
+        @AuraEnabled public npe03__Recurring_Donation__c recurringDonation;
+        @AuraEnabled public String frequency;
```

Comment:
what is the reason for having both recurringDonation full record and separate variables for the frequency and status

Reply:
Frequency and Status aren't custom fields from the npe03__Recurring_Donation__c object, they are calculated values ​​that will be displayed in the frontend component.
Having the npe03__Recurring_Donation__c object separated will allow to add future fields here without update the inner class format, and it will be useful in the fronted in the future. 
The idea is to have the pure npe03__Recurring_Donation__c object with the different fields used by the fronted, and the calculated values ​​to be displayed in the same object to be used by the table for each row.